src/main/java/com/example/crudapi/controller/LocationSearchController.java

```diff
@@ -24,4 +32,14 @@ public LocationSearchResponse findByCorner(@RequestParam("corner") @NotBlank Str
         Location location = locationSearchService.findByCorner(corner);
         return new LocationSearchResponse(location.convertToLocationSearchDto());
     }
+
+    @PostMapping("/location-search")
+    public ResponseEntity<Map<String, String>> createLocation(@RequestBody @Validated Locationform form, UriComponentsBuilder uriBuilder) {
+        Location location = locationSearchService.createLocation(form);
+        URI url = uriBuilder
+                .path("/location-search/" + location.getCorner())
```

Comment:
今更ですが、RESTful APIを実装するにあたってURLには動詞をなるだけ含めないのがよいです。
```suggestion
                .path("/locations/" + location.getCorner())
```

Reply:
`@GetMapping("/location-search")`も直したい。